src/box/sql/show.c

```diff
@@ -0,0 +1,425 @@
+		   const struct index *index)
+{
+	assert(index != NULL);
+	if (index->def->type != TREE || !index->def->opts.is_unique) {
+		sql_err_append(desc, "Index", index->def->name);
+		return;
+	}
+	for (uint32_t i = 0; i < index->def->key_def->part_count; ++i) {
+		uint32_t fieldno = index->def->key_def->parts[i].fieldno;
+		if (fieldno >= space->def->field_count) {
+			sql_err_append(desc, "Index", index->def->name);
+			return;
+		}
+		enum field_type field_type = space->def->fields[fieldno].type;
+		if (index->def->key_def->parts[i].type != field_type) {
+			sql_err_append(desc, "Index", index->def->name);
+			return;
+		}
+		uint32_t coll_id = space->def->fields[fieldno].coll_id;
+		if (index->def->key_def->parts[i].coll_id != coll_id) {
+			sql_err_append(desc, "Index", index->def->name);
+			return;
+		}
+	}
+
+	char *index_name = sql_escaped_name_new(index->def->name);
+	sql_desc_append(desc, ",\nCONSTRAINT %s UNIQUE(", index_name);
+	sql_xfree(index_name);
+	for (uint32_t i = 0; i < index->def->key_def->part_count; ++i) {
+		uint32_t fieldno = index->def->key_def->parts[i].fieldno;
+		const char *fmt = i == 0 ? "%s" : ", %s";
+		const char *name = space->def->fields[fieldno].name;
+		char *field_name = sql_escaped_name_new(name);
+		sql_desc_append(desc, fmt, field_name);
+		sql_xfree(field_name);
+	}
+	sql_desc_append(desc, ")");
+}
+
+char *
+sql_generate_show_create_table(uint32_t space_id)
+{
+	struct space *space = space_by_id(space_id);
+	assert(space != NULL);
+
+	struct sql_desc desc;
+	sql_desc_initialize(&desc);
+
+	if (space->def->field_count == 0) {
+		diag_set(ClientError, ER_UNSUPPORTED, "SQL",
+			 "space without format");
+		return NULL;
+	}
+
+	if (space->index_count == 0) {
+		diag_set(ClientError, ER_UNSUPPORTED, "SQL",
+			 "spaces without primary key");
+		return NULL;
+	}
+
+	const struct index *pk = space->index[0];
+	assert(pk->def->opts.is_unique);
+	if (pk->def->type != TREE) {
+		diag_set(ClientError, ER_UNSUPPORTED, "SQL",
+			 "non-TREE index type");
+		return NULL;
+	}
+
+	for (uint32_t i = 0; i < pk->def->key_def->part_count; ++i) {
+		uint32_t fieldno = pk->def->key_def->parts[i].fieldno;
+		if (fieldno >= space->def->field_count) {
+			diag_set(ClientError, ER_UNSUPPORTED, "SQL",
+				 "indexes with unnamed parts");
+			return NULL;
+		}
+		enum field_type field_type = space->def->fields[fieldno].type;
+		if (pk->def->key_def->parts[i].type != field_type) {
+			diag_set(ClientError, ER_UNSUPPORTED, "SQL",
+				 "indexes with part type not equal to "
+				 "corresponding field type");
+			return NULL;
+		}
+		uint32_t coll_id = space->def->fields[fieldno].coll_id;
+		if (pk->def->key_def->parts[i].coll_id != coll_id) {
+			diag_set(ClientError, ER_UNSUPPORTED, "SQL",
+				 "indexes with part collation not equal to "
+				 "corresponding field collation");
+			return NULL;
+		}
+	}
+
+	if (strcmp(space->engine->name, "memtx") != 0 &&
+	    strcmp(space->engine->name, "vinyl") != 0) {
```

Comment:
`space_is_memtx()` and `space_is_vinyl()` from `space.h`.

Reply:
Thanks! Done.